pkg/minikube/bootstrapper/bsutil/kverify/pod_ready.go

```diff
@@ -63,7 +66,14 @@ func WaitExtra(cs *kubernetes.Clientset, labels []string, timeout time.Duration)
 			}
 			if match || pod.Spec.PriorityClassName == "system-cluster-critical" || pod.Spec.PriorityClassName == "system-node-critical" {
 				if err := waitPodCondition(cs, pod.Name, pod.Namespace, core.PodReady, timeout); err != nil {
```

Comment:
how about instead of waiting till it fails and then check the error message to see if it is a "Termianted pod" how about at beggning only range through pods which are not Terminated ?


```	pods, err := cs.CoreV1().Pods(meta.NamespaceSystem).List(context.Background(), meta.ListOptions{})
	if err != nil {
		return fmt.Errorf("error listing pods in %q namespace: %w", meta.NamespaceSystem, err)
	}
```

Reply:
or as @ilya-zuyev noted 	`for _, pod := range pods.Items {` might have a condition we could check

Reply:
i think something like that (ie, 'skipping!') _was_ my original intention:
https://github.com/kubernetes/minikube/blob/357b5938a17f5f40d626fbf0e629cee88db30707/pkg/minikube/bootstrapper/bsutil/kverify/pod_ready.go#L137-L139
but it somehow got lost along the way (when i also added host checking)

i'll have a look...

Reply:
@medyagh @ilya-zuyev thanks for catching this!
it shouldn't have waited on non-running and non-pending pods because of:
https://github.com/kubernetes/minikube/blob/357b5938a17f5f40d626fbf0e629cee88db30707/pkg/minikube/bootstrapper/bsutil/kverify/pod_ready.go#L137-L139
and
https://github.com/kubernetes/minikube/blob/357b5938a17f5f40d626fbf0e629cee88db30707/pkg/minikube/bootstrapper/bsutil/kverify/pod_ready.go#L105-L108
then
https://github.com/kubernetes/minikube/blob/357b5938a17f5f40d626fbf0e629cee88db30707/pkg/minikube/bootstrapper/bsutil/kverify/pod_ready.go#L116-L118
would exit immediately (due to the returned error from checkCondition()), but i've missed that specific one in:
https://github.com/kubernetes/minikube/blob/357b5938a17f5f40d626fbf0e629cee88db30707/pkg/minikube/bootstrapper/bsutil/kverify/pod_ready.go#L72-L73
so i've corrected that in https://github.com/kubernetes/minikube/pull/11209/commits/6779c727ce03d1381f26383d3f9aa13bbe0f37f5 in https://github.com/kubernetes/minikube/pull/11209 to test it there first

Reply:
@ilya-zuyev @medyagh actually, this was ok - no need to change anything here as `wait` works as intended (it will skip if pod is gone or if the just-restarted-minion node is not yet ready (detailed explanation [above](https://github.com/kubernetes/minikube/pull/11106#discussion_r622586453) still holds)
i'll close this one in favour of https://github.com/kubernetes/minikube/pull/11209
/close